lib/client/doc.js

```diff
@@ -104,11 +104,20 @@ emitter.mixin(Doc);
 Doc.prototype.destroy = function(callback) {
   var doc = this;
   doc.whenNothingPending(function() {
-    doc.connection._destroyDoc(doc);
     if (doc.wantSubscribe) {
-      return doc.unsubscribe(callback);
+      doc.unsubscribe(function(err) {
+        if (err) {
+          if (callback) callback(err);
+          else this.emit('error', err);
```

Comment:
`this` should be doc. I'll just go ahead and merge this change and make the fix, since we're close.